File: ECnotes/Sem2/LivExamples/VB/Ch22WebParts/HalloweenCounter.ascx.vb

Partial Class HalloweenCounter
    Inherits System.Web.UI.UserControl

    Protected Sub Page_Load(ByVal sender As Object, _
            ByVal e As System.EventArgs) Handles Me.Load
        Dim dtmToday As Date = Date.Today
        Dim dtmHalloween As Date = New Date(Date.Today.Year, 10, 31)

        If dtmToday > dtmHalloween Then
            dtmHalloween = dtmHalloween.AddYears(1)
        End If

        Dim tsSpan As TimeSpan = dtmHalloween - dtmToday

        lblCounter.Text = "There are " & tsSpan.Days & " shopping days until Halloween."
    End Sub
End Class

File: ECnotes/Sem2/LivExamples/CS/Ch22WebParts/HalloweenCounter.ascx.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

public partial class HalloweenCounter : System.Web.UI.UserControl
{
    protected void Page_Load(object sender, EventArgs e)
    {
		DateTime today = DateTime.Today;

		DateTime halloween = new DateTime(DateTime.Today.Year, 10, 31);

		if (today > halloween)
			halloween = halloween.AddYears(1);

		TimeSpan span = halloween - today;

		lblCounter.Text = "There are " + span.Days + " shopping days until Halloween.";
    }
}
